tools/rosbag/src/player.cpp

```diff
@@ -106,8 +106,11 @@ Player::Player(PlayerOptions const& options) :
     // If we were given a list of topics to pause on, then go into that mode
     // by default (it can be toggled later via 't' from the keyboard).
     pause_for_topics_(options_.pause_topics.size() > 0),
+    pause_change_requested_(false),
+    requested_pause_state_(false),
     terminal_modified_(false)
 {
+  pause_service_ = node_handle_.advertiseService("pause", &Player::pauseCallback, this);
```

Comment:
I wasn't sure what the best route was here. The problem with a private node handle is that if you're running `rosbag play` outside a launch file, it will by default be given an anonymous name, and so calling the service is a pain (e.g., `rosservice call /play_1484077899473435649/pause`). As I couldn't envisage a use case wherein we'd be running more than one `rosbag play` instance, I figured a service in the global namespace would work.

However, I also can't really see using this outside of a launch file, and even if we did, we could assign a namespace via the command line. I'll move it back to the private node handle.